include/cvc5/cvc5_proof_rule.h

```diff
@@ -1382,10 +1388,12 @@ enum ENUM(ProofRule) : uint32_t
    * where t_1 and s_1 are constants of length one, or otherwise one side
    * of the equality is the empty sequence and t_1 or s_1 corresponding to
    * that side is the empty sequence.
+   * 
+   * This rule is used exclusively for sequences.
    *
    * \endverbatim
    */
-  EVALUE(CONCAT_CONFLICT),
+  EVALUE(CONCAT_CONFLICT_DEQ),
```

Comment:
I don't have a good name, plus I think its best to have this name since it can be seen as `CONCAT_CONFLICT` with an extra disequality (for the chars).